registry/registry_ui.go

```diff
@@ -480,22 +500,29 @@ func createUpdateNamespace(ctx *gin.Context, isUpdate bool) {
 		}
 
 		// Then check if the user has previlege to update
-		isAdmin, _ := web_ui.CheckAdmin(user)
+		belongsTo := false
+
 		if !isAdmin { // Not admin, need to check if the namespace belongs to the user
-			found, err := namespaceBelongsToUserId(ns.ID, user)
+			belongsTo, err = namespaceBelongsToUserId(ns.ID, user)
 			if err != nil {
 				log.Error("Error checking if namespace belongs to the user: ", err)
 				ctx.JSON(http.StatusInternalServerError, server_structs.SimpleApiResp{
 					Status: server_structs.RespFailed,
 					Msg:    "Error checking if namespace belongs to the user"})
 				return
 			}
-			if !found {
-				log.Errorf("Namespace not found for id: %d", ns.ID)
-				ctx.JSON(http.StatusNotFound, server_structs.SimpleApiResp{
-					Status: server_structs.RespFailed,
-					Msg:    "Namespace not found. Check the id or if you own the namespace"})
-				return
+			if !belongsTo {
+				// If the user doen's own the namespace, they can't edit it
+				// but if they have an access token generated by the private key
+				// that corresponds to the public key of the namespace they are trying to access,
+				//  we allow them to do it so that admin can fill out the rest of the registration if the user_id is empty
+				if accessToken == "" {
+					log.Errorf("Access denied from user %s for namespace with id=%d", user, id)
+					ctx.JSON(http.StatusNotFound, server_structs.SimpleApiResp{
+						Status: server_structs.RespFailed,
+						Msg:    "Namespace not found. Check the id or if you own the namespace"})
```

Comment:
Is this actually a "namespace not found" error, or is it a "permission denied" error? The message seems to suggest both.

Reply:
This is the part I'm not very confident about. I saw people returning 404 for objects that does not belong to a user (although it exists), (to confuse attackers?), but this will also somehow confuse our users a bit.

Reply:
I think we're probably safe to expose its under-construction existence here, especially if there are any other queries an attacker could perform to learn about it. Presumably, namespaces are public once they're registered anyway, so existence doesn't seem like it needs to be tightly protected.